src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -59,6 +61,30 @@ public override void Initialize (AnalysisContext context)
 				var invocationOperation = (IInvocationOperation) context.Operation;
 				ProcessInvocationOperation (context, invocationOperation);
 			}, OperationKind.Invocation);
+			context.RegisterSyntaxNodeAction (context => {
+				ProcessSyntaxNode (context);
+			}, SyntaxKind.GenericName);
+		}
+
+		private void ProcessSyntaxNode (SyntaxNodeAnalysisContext context)
+		{
+			if (context.SemanticModel.GetTypeInfo (context.Node).ConvertedType is not INamedTypeSymbol type)
+				return;
+
+			// The same INamedTypeSymbol can be called back resulting in duplicate warnings
+			// One way to filter is via the Node's Parent
+			if (context.Node.Parent is not ArgumentSyntax)
+				return;
```

Comment:
This feels dangerous - in that if there's a place where a named type can be used and it's not an argument syntax we would miss it - and potentially not report a warning. I'm definitely not an expert on this though.
@agocke - what would be the right way to avoid duplicate warnings while still not missing anything.

Reply:
I'm not sure I understand the duplication. Is it something like

```C#
var bad1 = new G<A>();
var bad2 = new G<A>();
```

where `G` is `G<[DAM]T>` and `A` doesn't have the requirement? In that case I would expect a warning in every place that the substitution occurs.

Reply:
This was due to the `nameof `identifier that is handled now